okonomiyaki/file_formats/_egg_info.py

```diff
@@ -0,0 +1,697 @@
+            return None
+        else:
+            return self._epd_legacy_platform.osdist
+
+    @property
+    def platform(self):
+        """
+        The legacy platform name (sys.platform).
+        """
+        if self._epd_legacy_platform is None:
+            return None
+        else:
+            return self._epd_legacy_platform.platform
+
+    @property
+    def metadata_version(self):
+        return self._metadata_version
+
+    @metadata_version.setter
+    def metadata_version(self, value):
+        self._metadata_version = value
+
+    @property
+    def metadata_version_info(self):
+        return _metadata_version_to_tuple(self._metadata_version)
+
+    def _to_dict(self):
+        raw_data = {
+            _TAG_NAME: self.name,
+            _TAG_VERSION: self.version,
+            _TAG_BUILD: self.build,
+            _TAG_ARCH: self.arch,
+            _TAG_PLATFORM: self.platform,
+            _TAG_OSDIST: self.osdist,
+            _TAG_PACKAGES: [str(p) for p in self.packages],
+            _TAG_PYTHON: self.python,
+            _TAG_PYTHON_TAG: self.python_tag,
+            _TAG_METADATA_VERSION: self.metadata_version
+        }
+
+        ret = {}
+        for k, v in raw_data.items():
+            if isinstance(v, six.string_types):
+                v = str(v)
+            ret[k] = v
+        return ret
+
+    def to_string(self):
+        """
+        Returns a string that is suitable for the depend file inside our
+        legacy egg.
+        """
+        template = _METADATA_TEMPLATES.get(self.metadata_version, None)
+        data = self._to_dict()
+
+        # This is just to ensure the exact same string as the produced by the
+        # legacy buildsystem
+        if len(self.packages) == 0:
+            data[_TAG_PACKAGES] = "[]"
+        else:
+            data[_TAG_PACKAGES] = "[\n{0}\n]". \
+                format("\n".join("  '{0}',".format(p)
+                       for p in self.packages))
+        return template.format(**data)
+
+
+class LegacySpec(HasTraits):
+    """
+    This models the EGG-INFO/spec content.
+    """
+    depend = Instance(LegacySpecDepend)
+    """
+    Models the spec/depend content
+    """
+
+    lib_depend = List()
+    """
+    List of freeform content
+    """
+    lib_provide = List()
+    """
+    List of freeform content
+    """
+
+    summary = Unicode()
+    """
+    Summary metadata of the egg.
+    """
+
+    @classmethod
+    def from_egg(cls, egg):
+        spec_depend = LegacySpecDepend.from_egg(egg)
+
+        data = {"depend": spec_depend}
+
+        with zipfile2.ZipFile(egg) as fp:
+            try:
+                lib_depend_data = fp.read(_SPEC_LIB_DEPEND_LOCATION).decode()
+                data["lib_depend"] = lib_depend_data.splitlines()
+            except KeyError:
+                pass
+
+        return cls(**data)
+
+    @property
+    def egg_name(self):
+        return "{0}-{1}-{2}.egg".format(self.depend.name,
```

Comment:
Yes, but that class will go away, so I don't think it is worth doing that change now. I did adapt `EggMetadata.egg_name` property though